app/javascript/admin/controllers/modal_controller.js

```diff
@@ -2,12 +2,37 @@ import { Controller } from 'stimulus';
 
 // eslint-disable-next-line no-restricted-syntax
 export default class ModalController extends Controller {
-  static classes = ['hidden'];
-  static targets = ['toggle'];
+  static values = {
+    contentId: String,
+    title: String,
+    size: String,
+  };
 
-  toggleModal() {
-    if (this.hasToggleTarget) {
-      this.toggleTarget.classList.toggle(this.hiddenClass);
-    }
+  async toggleModal() {
+    const [{ Modal }, { render, h }] = await Promise.all([
+      import('@crayons/Modal'),
+      import('preact'),
+    ]);
+
+    const modalRoot = document.createElement('div');
```

Comment:
Refactor (blocking): Keep the same `<div />` element instead of creating a new one everytime `toggleModal` is called. This can be a property on the modal instance.